File: ms-users/infrastructure/src/main/java/jva/cloud/config/CreateDependence.java
package jva.cloud.config;

import jva.cloud.ports.input.UserServicePort;
import jva.cloud.ports.output.UserPersistencePort;
import jva.cloud.service.UserService;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class CreateDependence {

    @Bean
    UserServicePort bindingToUserServicePort(UserPersistencePort userPersistencePort) {
        return new UserService(userPersistencePort);
    }
}

File: ms-products/infrastructure/src/main/kotlin/config/CreateDependence.kt
package jva.cloud.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import port.input.CategoryServicePort
import port.oatput.CategoryPersistencePort
import service.CategoryService

@Configuration
class CreateDependence {
    @Bean
    fun bindingToCategoryServicePort(categoryPersistencePort: CategoryPersistencePort): CategoryServicePort {
        return CategoryService(categoryPersistencePort = categoryPersistencePort)
    }
}